File: app/src/main/java/co/za/takeabyte/gradalot2020/promotions/itemdecoration/ItemDecorationPromotion.java
package co.za.takeabyte.gradalot2020.promotions.itemdecoration;

import android.content.Context;
import android.graphics.Rect;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import co.za.takeabyte.gradalot2020.R;

public class ItemDecorationPromotion extends RecyclerView.ItemDecoration {

    private int marginBig;

    public ItemDecorationPromotion(@NonNull final Context context) {
        if (context.getResources() != null) {
            marginBig = context.getResources().getDimensionPixelSize(R.dimen.margin_large);
        }
    }

    @Override
    public void getItemOffsets(@NonNull Rect outRect, @NonNull View view, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        outRect.left = marginBig;
        outRect.right = marginBig;
    }
}

File: app/src/main/java/co/za/takeabyte/gradalot2020/promotions/itemdecoration/ItemDecorationPromotion.kt
package co.za.takeabyte.gradalot2020.promotions.itemdecoration

import android.content.Context
import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import co.za.takeabyte.gradalot2020.R

// TODO: Margin on promotion items seems incorrect

class ItemDecorationPromotion(context: Context) : RecyclerView.ItemDecoration() {

    private val marginBig = context.resources.getDimensionPixelSize(R.dimen.margin_large)

    override fun getItemOffsets(
        outRect: Rect,
        view: View,
        parent: RecyclerView,
        state: RecyclerView.State
    ) {
    }
}